Quantum-Control-Applications-QuAM/Superconducting/README.md

```diff
@@ -0,0 +1,357 @@
+
+### calibration_graph
+This folder contains all the calibration scripts that can compose a qualibrate graph.
+The structure of the nodes is described below.
+
+### configuration
+The configuration folder contains the python scripts used to build the QUAM before starting the experiments.
+It contains three files whose working principles are explained in more details below:
+* [make_wiring.py](./configuration/make_wiring.py): create the port mapping between the control hardware (OPX+, Octave, OPX1000 LF fem, MW fem) and the quantum elements (qubits, resonators, flux lines...). 
+* [make_quam.py](./configuration/make_quam.py): create the state of the system based on the generated wiring and QUAM components and containing all the information necessary to calibrate the chip and run experiments. This state is used to generate the OPX configuration. 
+* [modify_quam.py](./configuration/modify_quam.py): update the parameters of the state programmatically based on defaults values (previous calibration, chip manufacturer specification...).
+
+### quam_libs
+This folder contains all the utility functions necessary to create the wiring or build the QUAM, as well as QUA macros and data processing tools:
+* [components](./quam_libs/components): this is where the QUAM root and custom QUAM components are defined. A set of basic QUAM components are already present, but advanced user can easily modify them or create new ones.
+* [lib](./quam_libs/lib): contains several utility functions for saving, fitting and post-processing data.
+* [quam_builder](./quam_libs/quam_builder): contains the main functions called in [machine.py](./quam_libs/quam_builder/machine.py) and used to generate the wiring and build the QUAM structure from it and the QUAM components declared in the [components](./quam_libs/components) folder. It also contains the [pulses.py](./quam_libs/quam_builder/pulses.py) file where the default qubits pulses are defined.
+
+## How to generate the QuAM
+Before starting to run experiments, it is necessary to build the Quantum Abstract Machine ([QuAM](https://github.com/qua-platform/quam)) for the desired 
+qubit chip architecture. More details about QUAM itself can be found in the [QuAM documentation](https://qua-platform.github.io/quam/).
+
+The process can be divided into three sections:
+1. First one needs to create the wiring specifying which quantum element (qubit, resonator, flux line...) is connected to 
+which OPX channel. 
+2. Then, the QuAM components can be derived from the resulting wiring.json file. 
+The attributes (methods or parameters) of these components can be customized at will. 
+3. Finally, a file called state.json, containing all the desired parameters that describe the full state of the system 
+will be created.
+
+### [1. Define the wiring](./configuration/make_wiring.py)
+The wiring is generated from a python script called [make_wiring.py](./configuration/make_wiring.py).
```

Comment:
@TheoLaudatQM there is actually no make_wiring.py file, but there are three make_wiring for the different instruments set ups

Reply:
Good point, I just fixed it